library/posture.c

```diff
@@ -165,29 +237,219 @@ void ziti_send_posture_data(struct ziti_ctx *ztx) {
     free(domainInfo);
     free(osInfo);
     free(macInfo);
-    //no free(procInfo), free'ed in map
-    ZITI_LOG(VERBOSE, "done sending posture data");
+    //no free(procInfo), free'ed in model_map_clear which calls free on values
+
+    ziti_send_pr(ztx);
+}
+
+static void ziti_collect_pr(ziti_context ztx, const char *pr_obj_key, char *pr_obj, int pr_obj_len) {
+    struct pr_info *current_info = malloc(sizeof(pr_info));
+
+    current_info->obj = pr_obj;
+    current_info->length = pr_obj_len;
+    current_info->should_send = true;
+    current_info->errored = false;
+
+    //selectively send if we don't have to send every time, we have previous responses to calculate against, and we aren't being forced to send
+    if (!ztx->posture_checks->must_send_every_time && ztx->posture_checks->previous_responses != NULL && ztx->posture_checks->must_send == false) {
+        struct pr_info *prev_info;
+        prev_info = model_map_get(ztx->posture_checks->previous_responses, pr_obj_key);
+
+        if (prev_info != NULL) {
+            if (prev_info->errored) {
+                current_info->should_send = true;
+            } else {
+                int info_cmp = strcmp(prev_info->obj, pr_obj); //assumes obj marshaling is deterministic
+
+                if (info_cmp == 0) { //same so don't sendd
+                    current_info->should_send = false;
+                } else {
+                    ZITI_LOG(DEBUG, "will send pr, comparison result for pr objects %d != 0, objects: %s -and- %s",
+                             info_cmp, prev_info->obj, pr_obj);
+                }
+            }
+        } else {
+            ZITI_LOG(DEBUG, "will send pr, prev object is null: %s", pr_obj);
+        }
+    } else {
+        ZITI_LOG(DEBUG, "will send pr, must_send == true or no previous responses: %s", pr_obj);
+    }
+
+    model_map_set(ztx->posture_checks->current_responses, pr_obj_key, current_info);
+}
+
+static void ziti_pr_post_bulk_cb(void *obj, ziti_error *err, void *ctx) {
+    ziti_context ztx = ctx;
+    if (err != NULL) {
+        ZITI_LOG(ERROR, "error during bulk posture response submission (%d) %s", err->http_code, err->message);
+        ztx->posture_checks->must_send = true; //error, must try again
+        if (err->http_code == 404) {
+            ztx->posture_checks->bulk_supported = false;
```

Comment:
I would like to consolidate controller features into one place to be able to reset them when controller version changes (and not requiring re-init/restart)

ziti_context already has one flag here https://github.com/openziti/ziti-sdk-c/blob/main/inc_internal/zt_internal.h#L181

Reply:
Is the only thing you want for me to move my `bool` there ?

Reply:
Move and swapped it to `no_bulk_posture_response_api` to match `no_service_updates_api`